sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -82,7 +84,15 @@ case class HadoopFsRelation(
     }
   }
 
-  override def sizeInBytes: Long = location.sizeInBytes
+  override def sizeInBytes: Long = {
+    val size = location.sizeInBytes * hadoopFSSizeFactor
+    if (size > Long.MaxValue) {
```

Comment:
I think this branch can be removed? `Long.MaxValue` is returned when converting a double value larger than `Long.MaxValue`.